fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/JoinedRecordTypeTest.java

```diff
@@ -103,52 +108,13 @@ void loadSyntheticRecord() throws ExecutionException, InterruptedException {
         }
     }
 
-    @Test
-    void loadSyntheticRecordFailsMissingConstituent() throws ExecutionException, InterruptedException {
-        final Tuple joinedPrimaryKey;
-        final FDBStoredRecord<Message> simpleRecord;
-        final FDBStoredRecord<Message> otherRecord;
-        try (FDBRecordContext context = openContext()) {
-            createOrOpenRecordStore(context, baseMetaData(addJoinedType()));
-
-            Tuple syntheticRecordTypeKey = recordStore.getRecordMetaData()
-                    .getSyntheticRecordType(JOINED_RECORD_NAME)
-                    .getRecordTypeKeyTuple();
-            simpleRecord = recordStore.saveRecord(createSimpleRecord(100, 10));
-            otherRecord = recordStore.saveRecord(createOtherRecord(101, 11));
-            joinedPrimaryKey = Tuple.from(
-                    syntheticRecordTypeKey.getItems().get(0),
-                    simpleRecord.getPrimaryKey().getItems(),
-                    otherRecord.getPrimaryKey().getItems());
-
-            context.commit();
-        }
-
-        try (FDBRecordContext context = openContext()) {
-            createOrOpenRecordStore(context, baseMetaData(addJoinedType()));
-
-            recordStore.deleteRecord(otherRecord.getPrimaryKey());
-
-            context.commit();
-        }
-
-        try (FDBRecordContext context = openContext()) {
-            createOrOpenRecordStore(context, baseMetaData(addJoinedType()));
-            // Default policy (ERROR) should fail
-            final ExecutionException exception = assertThrows(ExecutionException.class, () -> recordStore.loadSyntheticRecord(joinedPrimaryKey).get());
-            assertEquals(RecordDoesNotExistException.class, exception.getCause().getClass());
-            // RETURN policy returns the shell of the synthetic record with no constituents
-            final FDBSyntheticRecord syntheticRecord = recordStore.loadSyntheticRecord(joinedPrimaryKey, IndexOrphanBehavior.RETURN).get();
-            assertEquals(0, syntheticRecord.getConstituents().size());
-            // SKIP policy returns null in case of missing constituents
-            assertNull(recordStore.loadSyntheticRecord(joinedPrimaryKey, IndexOrphanBehavior.SKIP).get());
-
-            context.commit();
-        }
+    static Stream<Arguments> twoBooleans() {
```

Comment:
`twoBooleans` is a bit vague. Outside of the implementation, I would assume it would have all 4, but it's skipping `false, false`, because that's tested above. Perhaps renaming this to just the name of the test, making it tightly coupled to the test itself. Or you could use `@CsvSource`

Reply:
Sure, changed to CsvSource